src/box/index_def.c

```diff
@@ -107,14 +108,23 @@ index_def_new(uint32_t space_id, uint32_t iid, const char *name,
 	}
 	def->name = strndup(name, name_len);
 	if (def->name == NULL) {
-		index_def_delete(def);
 		diag_set(OutOfMemory, name_len + 1, "malloc", "index_def name");
-		return NULL;
+		goto error;
```

Comment:
While you are at it, please make `index_def_new` use `xstrdup` and `xmalloc` and remove all checks if it returns `NULL`.

Reply:
Fixed.